### ISSUE TITLE
salsa_kernel.cu:67: undefined reference to `TestKernel::TestKernel()

### ISSUE THREAD

**Issue by drmike8888** [cbuchner1/CudaMiner#18]
This error happens because the test_kernel.cu file is not compiled (there is no .o).  Brute force compiling test_kernel then gives a .o, but it is not listed as part of the link list:
g++  -g -O2 -pthread -L/usr/local/cuda/lib64  -o cudaminer cudaminer-cpu-miner.o cudaminer-util.o cudaminer-sha2.o cudaminer-scrypt.o salsa_kernel.o spinlock_kernel.o legacy_kernel.o fermi_kernel.o titan_kernel.o -L/usr/lib/x86_64-linux-gnu -lcurl -Wl,-Bsymbolic-functions -Wl,-z,relro compat/jansson/libjansson.a -lpthread  -lcudart -fopenmp  
salsa_kernel.o: In function `validate_config(char*, int&, int&, KernelInterface**, cudaDeviceProp*)':
/home/drmike/Litecoin/CudaMiner-master/salsa_kernel.cu:67: undefined reference to`TestKernel::TestKernel()'

Brute force adding test_kernel.o in the list allows a full compile/link and run.  Is there an easy way to fix the Makefile?

**Comment by cbuchner1** [comment]
Does running ./autgen.sh fix it? I made changes to Makefile.am, but not to Makefile. And I do not usually compile on Linux (my development is Windows based at the moment) so I rarely update the ./configure script and Makefile on github.

**Comment by drmike8888** [comment]
I will try that when I get home from work.  Thanks for the fast reply!

**Comment by drmike8888** [comment]
OK, I had to change it so ./aclocal.m4 was referenced, and make that 
executable, and then I got this error:

./aclocal.m4: line 14: syntax error near unexpected token 
`[AC_AUTOCONF_VERSION],'
./aclocal.m4: line 14:`m4_ifndef([AC_AUTOCONF_VERSION],'

Don't worry about it unless some other Ubuntu person shows up.  My machine 
could only generate 152kH/s which amounts to .02 Lite-coins per day given 
how the network to my house is.  So I'm not going to pursue that any more.

But I am really impressed with your code and I intend to study it so I can 
better utilize my Cuda engine for doing quantum mechanics calculations. 
The Nvidia cards are optimized for floating point, not bit flipping.  Now 
that I've got Cuda-5.5, and I know it works, I can go have my own fun.

Excellent coding job - you have a lot to be proud of!
Patience, persistence, truth,
Dr. mike
